File: LC/0575_Distribute_Candies/solution.cpp
// link of the question: https://leetcode.com/problems/distribute-candies/

class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        unordered_set<int> unique_candy_type(candyType.begin(), candyType.end());

        if (unique_candy_type.size() <= candyType.size() / 2) {
            return unique_candy_type.size();
        } else {
            return candyType.size() / 2;
        }
    }
};

File: LC/0575_Distribute_Candies/solution.py
# link of the question: https://leetcode.com/problems/distribute-candies/

class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        unique_candy_type = set(candyType)
        if len(unique_candy_type) <= len(candyType) / 2:
            return len(unique_candy_type)
        else:
            return len(candyType) // 2